modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.jsx

```diff
@@ -27,6 +31,12 @@ const Layout = () => {
 		}
 	}, [accountKey]);
 
+	if (userProjectAccess) {
+		if (userProjectAccess.denyAccess || !userProjectAccess.hasProjectAccess) {
+			return <ProjectErrorMessage />;
+		}
+	}
```

Comment:
```suggestion
	if (userProjectAccess?.denyAccess || !userProjectAccess?.hasProjectAccess) {
		return <ProjectErrorMessage />;
	}
```

Reply:
Just out of curiosity, is there a reason why alot of the frontend code like to use the ? syntax to check for objects or undefined objects? I find that we use it alot.

Reply:
Also, in this file, the userProjectAccess will sometimes be undefined that if I use this check, it will flash the Project Error Message page before all the other pages finish loading. I added the ` if (userProjectAccess)` so it would only load the error page if context threw an error and set the denyAccess value or if the user really does or does not have permissions.

Reply:
Got it. So you can keep that if.

> Just out of curiosity, is there a reason why alot of the frontend code like to use the ? syntax to check for objects or undefined objects? I find that we use it alot.
It's another syntax pattern.

Reply:
@jairmedeiros  Can you show us where this syntax pattern is being confirmed that it's standard? I was talking to Kyle a few weeks ago and we were noticing that there's alot of optional ? (optional undefined checks) where it doesn't logically make sense to use it. Generally, I think both frontend developers and backend developers need to know when their variables are either undefined or defined properly. To add ? to the code feels like a "catch call" which could potentially hide bugs we should be fixing.